pkg/webhook/membercluster/membercluster_validating_webhook.go

```diff
@@ -22,26 +25,47 @@ var (
 )
 
 type memberClusterValidator struct {
+	client  client.Client
 	decoder webhook.AdmissionDecoder
 }
 
 // Add registers the webhook for K8s bulit-in object types.
 func Add(mgr manager.Manager) error {
 	hookServer := mgr.GetWebhookServer()
-	hookServer.Register(ValidationPath, &webhook.Admission{Handler: &memberClusterValidator{admission.NewDecoder(mgr.GetScheme())}})
+	hookServer.Register(ValidationPath, &webhook.Admission{Handler: &memberClusterValidator{client: mgr.GetClient(), decoder: admission.NewDecoder(mgr.GetScheme())}})
 	return nil
 }
 
 // Handle memberClusterValidator checks to see if member cluster has valid fields.
-func (v *memberClusterValidator) Handle(_ context.Context, req admission.Request) admission.Response {
+func (v *memberClusterValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	mcObjectName := types.NamespacedName{Name: req.Name, Namespace: req.Namespace}
+	klog.V(2).InfoS("Validating webhook handling member cluster", "operation", req.Operation, "memberCluster", mcObjectName)
+
+	if req.Operation == admissionv1.Delete { // Will reject the requests whenever the serviceExport is not deleted
```

Comment:
I have some random thoughts, these are not blocker per se, just something that occurred to me when reading the code:

Reply:
We already have a webhook that guards against deletion attempts on Azure Fleet Manager from users (or any source other than RP, AKS support, etc.). So technically speaking the added logic is added to prevent RP (in most cases) from triggering the deletion. How would these components interact I guess? The concern is mostly that users won't see the error message that reminds them to delete service exports.

Reply:
For upstream they have full control so we generally assume that protection is not 100% a must.

Reply:
In the RP side, we should surface this error to the user. Will test that.